ts-client/src/dlmm/index.ts

```diff
@@ -4146,6 +4150,43 @@ export class DLMM {
     };
   }
 
+  /**
+   * Initializes bin arrays for the given bin array indexes if it wasn't initialized.
+   *
+   * @param {BN[]} binArrayIndexes - An array of bin array indexes to initialize.
+   * @param {PublicKey} funder - The public key of the funder.
+   * @return {Promise<TransactionInstruction[]>} An array of transaction instructions to initialize the bin arrays.
+   */
+  public async initializeBinArrays(binArrayIndexes: BN[], funder: PublicKey) {
+    const ixs: TransactionInstruction[] = [];
+
+    for (const idx of binArrayIndexes) {
+      const [binArray] = deriveBinArray(
+        this.pubkey,
+        idx,
+        this.program.programId
+      );
+
+      const binArrayAccount =
+        await this.program.provider.connection.getAccountInfo(binArray);
+
+      if (binArrayAccount === null) {
+        ixs.push(
```

Comment:
should we map many initializeBinArray in 1 transaction?

Reply:
I wish to let the user handle the blockhash for the transaction, therefore I return an array of instructions.